### PR TITLE
Data type should be always be snake case.

### PR DESCRIPTION
**Pull request by dompuiu**

### CODE REVIEW

**Comment by remear** [comment]
Does this also apply the case change as expected to types under relationship requests?

File: test/test_params_case_translator.rb [hunk 1]
```diff
@@ -86,4 +86,20 @@ def test_data_case_camel_complex
...
+      {
+        'HTTP_X_API_DATA_CASE' => 'camel-lower',
+        'action_dispatch.request.request_parameters' => params
+      }
+    )
+    status, headers, body = @translator.call(env)
+    assert_equal 200, status
+    response_params = body.first
+    assert_equal 'data_element', response_params['data']['type']
+end
```
Comment (remear) [inline test/test_params_case_translator.rb]: Is this an extra `end`?

I'd also like to see more comprehensive tests for this so we can guard against unexpected behavior. Perhaps additional payloads that have relationships and a kay with a value that has keys and values that shouldn't be touched by the case translator.

**Comment by dompuiu** [comment]
The relationships didn't work. I added a test for those and make them work. Please take a look.

File: lib/superintendent/request/params_case_translator.rb [hunk 2]
```diff
@@ -27,10 +27,31 @@ def underscored_keys(value)
       when Array
         value.map { |v| underscored_keys(v) }
       when Hash
+        if value['data'].present?
+          transform_type_to_snake_case(value['data'])
+        end
```
Comment (spencerroan) [inline lib/superintendent/request/params_case_translator.rb]: `underscored_keys` is already recursive and handles arrays and hashes.
 You can just do this one line here instead of adding a function.
```
  value['type'] = underscored_key(value['type']) if value.has_key?('type')
```

**Review by spencerroan — CHANGES_REQUESTED** [review]

File: lib/superintendent/request/params_case_translator.rb (hunk 2 above)
Comment (dompuiu) [reply lib/superintendent/request/params_case_translator.rb]: That works and I like it. But it will change any `type` property it finds. I'm fine with that, but I was under the impression that we need to touch only the type properties that have a `data` property as parent. If this is not the case, I can gladly change the PR to contain the code proposed by @spencerroan.

Comment (spencerroan) [reply lib/superintendent/request/params_case_translator.rb]: Good call. We should add a test similar to the first one where we don't expect type to change with something like:

 { 'data' => { 'type' => 'dataElements', 'settings' => {'type' => 'Foo'} } }

Then we can still do something similar to what I suggested.
```
if value.dig('data', 'type').present?
  value['data']['type'] = underscored_key(value['data']['type'])
end
```

**Comment by dompuiu** [comment]
@remear I pushed a new more compact version. And added some extra tests.

**Comment by spencerroan** [comment]
This looks more compact to me, I like it. Also the tests test the positive and negative behaviors. 👍
